### ISSUE TITLE
Large files downloads eat up all memory for S3

### ISSUE THREAD

**Issue by adejanovski** [thelastpickle/cassandra-medusa#52]
As reported in #46, restoring backups with large files can end up failing because of memory usage.
It seems that the simple `blob.download()` method of libcloud tries to read the data in memory before writing it to disk.

As it was done for uploads, we need to rely on awscli for now in order to use multi part downloads on large files.

**Comment by Kami** [comment]
An alternative should also be to utilize ``driver.download_object_as_stream`` method which shouldn't buffer the whole object in memory (same as ``driver.upload_object_via_stream`` method for the upload path).

**Comment by adejanovski** [comment]
Hi @Kami,
thanks, that was one of [my options](https://github.com/thelastpickle/cassandra-medusa/issues/46#issuecomment-572418700).
I went with the awscli for now since I already had the code in place for uploads.

I've seen that libcloud 3.0 will drop support for python < 3.5, which will make it easier to handle multi part transfers properly as we'll be able to use concurrency more easily than if we had to support py27 as well. I intend to contribute that in libcloud because I don't like to rely on a 3rd party command line tool for transfers, but performance wise we had no immediate choice.
